src/Data/List/Properties.agda

```diff
@@ -527,6 +670,12 @@ module _ {P : Pred A p} {f : A → A → A} where
 ------------------------------------------------------------------------
 -- foldl
 
+foldl-cong : ∀ {f g : B → A → B} {d e : B} →
+             (∀ x y → f x y ≡ g x y) → d ≡ e →
```

Comment:
Having a proof of `d ≡ e` isn't very consistent with the other cong proofs above right? They omit propositional equalities, and only look at functional equalities.

Reply:
yeah, that was my thought as well, but I preferred to mirror the structure of the existing `foldr-cong` proof that does accept `d ≡ e`. can be changed if you prefer

Reply:
see also [my comment](https://github.com/agda/agda-stdlib/issues/2358#issuecomment-2051480712) on #2358

Reply:
... and for having introduced a lemma whose name/type should be (downstream) deprecated in favour of the 'correct' version... oh dear!

Reply:
@MatthewDaggitt should i remove the `d ≡ e` or leave it in place to mirror `foldr-cong`? happy to do what you all prefer here

Reply:
Please could remove it? I think in the long run we want to try and remove it from the existing proofs as well.

Reply:
done. `d ≡ e` was replaced by an (also explicit) `y` argument